### ISSUE TITLE
[Meteor@1.3-rc.2] HTTP.post throwing errors when data is added

### ISSUE THREAD

**Issue by patrickml** [meteor/meteor#6505]
Whenever i use the HTTP package and pass in `{ data : {} }` into the second parameter an error is thrown.

the error can be seen in this photo https://www.dropbox.com/s/didyehth8u0n3v5/Screenshot%202016-03-16%2014.50.40.png?dl=0

```js
// Example useage
HTTP.post('http://localhost:4000/api/incoming/im',{ data : {} }, (err, res) => { console.log(err, res) })
```
// Error
```
httpcall_client.js:189 OPTIONS http://localhost:4000/api/incoming/im HTTP.call @ httpcall_client.js:189HTTP.post @ httpcall_common.js:61(anonymous function) @ VM1365:2InjectedScript._evaluateOn @ VM1358:875InjectedScript._evaluateAndWrap @ VM1358:808InjectedScript.evaluate @ VM1358:664
```

**Comment by M4v3R** [comment]
This is not an error in Meteor. You are probably using a wrong port number in your code (Meteor by default listens on port 3000 in development). According to [CORS specification](https://developer.mozilla.org/en-US/docs/Web/HTTP/Access_control_CORS#Overview) it is not allowed to send an XHR request to a server or another domain and/or port number, unless that server responds properly to an OPTIONS request that the browser sends first.

**Comment by zol** [comment]
Thanks @M4v3R, I believe you're right.

**Comment by patrickml** [comment]
@M4v3R you would think thats the issue, however I was running 2 meteor apps one on port 3000, and one on port 4000. In order to keep my API the same and allow for diversity amongst other applications to be able to connect to my app running on port 4000 I decided it was best to not use DDP and instead go with a traditional endpoint. My other applications (Non-meteor) were able to communicate just fine. Also this issue only arose after upgrading from rc-1 to rc-2. 

However, unexpectedly things started working again. -- Im let to believe it was an unexpected issue with chrome as i did not update my API whatsoever.

I had updated from rc-1 to rc-2 without closing my browser tab, not sure if this had anything to do with it as once i restarted chrome things worked fine again.

@zol might want to keep an eye on this as an edge case issue

Id like to thank both of you for your time looking into this issue.
